File: src/main/java/com/salon24/crawler/UrlPartitioner.java
package com.salon24.crawler;

import java.util.List;

class UrlPartitioner {

    PartitionedUrls paritionUrls(List<String> urls) {
        PartitionedUrls partitionedUrls = new PartitionedUrls();

        for (String url : urls) {
            if (isArticle(url)) {
                partitionedUrls.getArticles().add(url);
            } else if (isTag(url)) {
                partitionedUrls.getTags().add(url);
            } else if (isUser(url)) {
                partitionedUrls.getUsers().add(url);
            } else {
                partitionedUrls.getOther().add(url);
            }
        }

        return partitionedUrls;
    }

    private boolean isArticle(String url) {
       return url.matches("https://www.salon24.pl/u/(.+)/(\\d+),(.+)");
    }

    private boolean isTag(String url) {
        return url.matches("https://www.salon24.pl/k/(\\d+),(.+)");
    }

    private boolean isUser(String url) {
        return url.matches("https://www.salon24.pl/u/(.+)/");
    }
}

File: src/main/kotlin/com/salon24/crawler/UrlPartitioner.kt
package com.salon24.crawler

import org.springframework.stereotype.Component

@Component
class UrlPartitioner {
    companion object {
        private val IS_ARTICLE_REGEX = Regex("https://www.salon24.pl/u/(.+)/(\\d+),(.+)")
        private val IS_TAG_REGEX = Regex("https://www.salon24.pl/k/(\\d+),(.+)")
        private val IS_USER_REGEX = Regex("https://www.salon24.pl/u/(.+)/")
    }

    fun partition(urls: List<String>): PartitionedUrls {
        val partitionedUrls = PartitionedUrls()

        urls.forEach { url ->
            when {
                url.matches(IS_ARTICLE_REGEX) -> partitionedUrls.articles.add(url)
                url.matches(IS_TAG_REGEX) -> partitionedUrls.tags.add(url)
                url.matches(IS_USER_REGEX) -> partitionedUrls.users.add(url)
                else -> partitionedUrls.others.add(url)
            }
        }

        return partitionedUrls
    }

}